src/Providers/ProviderRegistry.php

```diff
@@ -204,7 +247,20 @@ public function getProviderModel(
 
         // Use static method from ProviderInterface
         /** @var class-string<ProviderInterface> $className */
-        return $className::model($modelId, $modelConfig);
+        $modelInstance = $className::model($modelId, $modelConfig);
+
+        if ($modelInstance instanceof WithHttpTransporterInterface) {
+            $modelInstance->setHttpTransporter($this->getHttpTransporter());
+        }
+
+        if ($modelInstance instanceof WithRequestAuthenticationInterface) {
+            $requestAuthentication = $this->getProviderRequestAuthentication($className);
+            if ($requestAuthentication !== null) {
+                $modelInstance->setRequestAuthentication($requestAuthentication);
+            }
+        }
```

Comment:
Per our chat, let's do this:
1. Create a `ProviderRegistry::bindModelDependencies($model)` method that handles this work
2. Make sure that the model instance in the `PromptBuilder` uses this via the `AiClient`
3. Update the `usingModel` method to support the same parameters as this **as well** (😉) as the existing model instance syntax

Reply:
Failing tests, but otherwise this is done